testutils/echo.go

```diff
@@ -32,17 +32,58 @@ import (
 )
 
 // CallEcho calls the "echo" endpoint from the given src to target.
-func CallEcho(src *tchannel.Channel, targetHostPort, targetService string, args *raw.Args) error {
+func CallEcho(
+	src *tchannel.Channel,
+	targetHostPort string,
+	targetService string,
+	args *raw.Args,
+) error {
+	return CallEchoWithContext(
+		nil,
+		src,
+		targetHostPort,
+		targetService,
+		args,
+	)
+}
+
+// CallEchoWithContext calls the "echo" endpoint from the given src to target,
+// using any deadline within the given context.Context.
+func CallEchoWithContext(
+	ctx context.Context,
+	src *tchannel.Channel,
+	targetHostPort string,
+	targetService string,
+	args *raw.Args,
+) error {
 	if args == nil {
 		args = &raw.Args{}
 	}
 
-	ctx, cancel := tchannel.NewContextBuilder(Timeout(300 * time.Millisecond)).
+	timeout := time.Second
+	if ctx != nil {
```

Comment:
why support `ctx == nil` as a valid argument (which isn't common for context usages in go), when `CallEcho` can be used instead?